Cabal/Distribution/Parsec/Types/Field.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE DeriveFunctor #-}
+module Distribution.Parsec.Types.Field (
+    -- * Cabal file
+    Field (..),
+    fieldAnn,
+    FieldLine (..),
+    SectionArg (..),
+    sectionArgAnn,
+    -- * Name
+    Name (..),
+    mkName,
+    getName,
+    nameAnn,
+    ) where
+
+import Prelude ()
+import Distribution.Compat.Prelude
+
+import Data.ByteString (ByteString)
+
+import qualified Data.Char as Char
+import qualified Data.ByteString.Char8 as B
+
+-------------------------------------------------------------------------------
+-- Cabal file
+-------------------------------------------------------------------------------
+
+data Field ann
```

Comment:
Have to add note: it might feel that we don't need `IfElseBlock`, but it's separate form section as there could be follow-up "else". OTOH we could handle those in actual cabal file parser.

I'm not sure which solution is clearer. E.g. for config or BuildHooksInfo `IfElseBlock` is fully unnecessary, but we cannot know that statically...